polygeist/tools/cgeist/Lib/CodeGenTypes.cc

```diff
@@ -1282,14 +1300,12 @@ mlir::Type CodeGenTypes::getMLIRType(clang::QualType QT, bool *ImplicitRef,
       // specifically handle this case by unwrapping the clang-adjusted
       // type, to get the corresponding ConstantArrayType with the full
       // dimensions.
-      if (MemRefFullRank) {
-        clang::QualType OrigTy = DT->getOriginalType();
-        if (OrigTy->isConstantArrayType()) {
-          SmallVector<int64_t, 4> Shape;
-          clang::QualType ElemTy;
-          getConstantArrayShapeAndElemType(OrigTy, Shape, ElemTy);
-          return mlir::MemRefType::get(Shape, getMLIRType(ElemTy));
-        }
+      if (MemRefFullRank && OrigTy->isConstantArrayType()) {
+        SmallVector<int64_t, 4> Shape;
+        clang::QualType ElemTy;
+        getConstantArrayShapeAndElemType(OrigTy, Shape, ElemTy);
+        return mlir::MemRefType::get(
+            Shape, getMLIRTypeForMem(ElemTy, ImplicitRef, AllowMerge));
```

Comment:
The original code uses the default value for `ImplicitRef` and `AllowMerge` instead of the values that passed in, do we want to keep that?
```suggestion
            Shape, getMLIRTypeForMem(ElemTy));
```

Reply:
I must've introduced those arguments by mistake. Fixed already.